DEVELOPER.md

```diff
@@ -70,9 +70,55 @@ pip-sync requirements\win-amd64-py3.11-requirements-dev.txt
 If the requirements file you used is updated, or if you want to switch to another requirements file (you need access to the Tensorflow library, for example), just run `pip-sync` again using the appropriate filename.
 It will install, upgrade, and uninstall all packages accordingly and ensure that you have a consistent environment.
 
-### Frontend development setup
-
-For instructions on how to prepare the frontend development environment, [see the src/frontend/README.md file](src/frontend/README.md)
+### Local Development setup (without containers)
+- Paths assume you are at the root of the cloned repository at https://github.com/usnistgov/dioptra
+- [Install redis](https://redis.io/docs/latest/operate/oss_and_stack/install/install-redis/)
+- Create a work directory for files `mkdir -p ~/dioptra/deployments/dev`
+- [Create a python virtual environment](#setting-up-the-python-virtual-environment)
+
+- The following describes commands to execute in four different terminal windows:
+  1. Flask Terminal
+    - Environment variables that must be set for flask:
+```bash
+DIOPTRA_RESTAPI_DEV_DATABASE_URI="sqlite:////home/<username>/dioptra/deployments/dev/dioptra-dev.db"
+DIOPTRA_RESTAPI_ENV=dev
+DIOPTRA_RESTAPI_VERSION=v1
+```
+    N.B.: replace <username> with your username. On some systems the home path may also be different. Verify the expansion of '~' with the `pwd` command while in the appropriate directory.
+
+    - Activate the python environment set-up in prior steps
+    - `dioptra-db autoupgrade`
+    - `flask run`
+
+  2. Frontend UI Terminal
+    - Commands to get a Frontend running:
+```bash
+cd src/fronted
+npm install
+npm run dev
+```
+
+  3. Redis Terminal
+    - `redis-server`
+
+  4. Dioptra Worker
+    - Starting a Dioptra Worker requires the following environment variables:
+```
+DIOPTRA_WORKER_USERNAME="dioptra-worker"  # This must be a registered user in the Dioptra app
```

Comment:
Do we want to call this out as a separate step? I feel like it might be easy to miss in a comment. We could even provide the curl command to register the user.

```sh
curl http://localhost:5000/api/v1/users/ -X POST --data-raw "{\"username\": \"$DIOPTRA_WORKER_USERNAME\",  \"email\": \"dioptra-worker@localhost\", \"password\": \"$DIOPTRA_WORKER_PASSWORD\", \"confirmPassword\": \"$DIOPTRA_WORKER_PASSWORD\"}"
```

Reply:
added some text to remind to do this step as well as create the queue